contracts/Hub.sol

```diff
@@ -0,0 +1,140 @@
+        authorizedAddresses.push(addr);
+    }
+
+
+    /**
+     * @notice mints synthAssets to a specific address
+     * @param receiver address to which token gets minted
+     * @param amount amount of token that gets minted
+     * @param tickerSymbol identifier of which token gets minted
+     */
+    function mintSynthAsset(address receiver, uint256 amount, string memory tickerSymbol) public onlyAuthorizedAddresses{
+        ISyntheticAsset(tickersymbolToSynthContractAddress[tickerSymbol]).mint(receiver, amount); 
+    }
+
+    /**
+     * @notice sets the Minter contract address
+     * @param _minterAddress address of Minter contract
+     */
+    function setMinterContractAddress(address _minterAddress, string memory tickerSymbol) public onlyOwner{
+        tickersymbolToSynthContractAddress[tickerSymbol] = _minterAddress;
+    }
+
+    /**
+     * @notice sets the Fund contract address
+     * @param _fundAddress address of Fund contract
+     */
+    function setCollateralFundsContract(address _fundAddress) public onlyOwner{
+        fundContract = ICollateralFunds(_fundAddress);
+    }
+
+    /**
+     * @notice sets the Storage contract address
+     * @param _storageAddress address of Storage contract
+     */
+    function setStorageContract(address _storageAddress) public onlyOwner{
+        storageContract = IStorage(_storageAddress);
+    }
+
+    /**
+     * @notice checks the collateral ratio of an address.
+     * @dev for all assets combined. No individual / isolated positions for now.
+     * @dev price checking of colllateral too -> volatile collateral / depegging of stable collateral
+     * @param addr address of user whom collateral ratio is to be checked.
+     * @param collateralTickerSymbol identifier of token used for collateral#
+     * @return collateral ratio
+     */
+        function getCollateralRatioByAddress(address addr, string memory collateralTickerSymbol) public view returns (uint256) {
+         
+         //Check amount funded
+         uint256 amountFunded = fundContract.getCollateralByAddress(msg.sender);
+         uint256 collateralPrice = storageContract.getAssetPrice(collateralTickerSymbol);
+         uint256 collateralValue = amountFunded * collateralPrice;
+
+         //Check assets minted
+         string[] memory assetsMinted = storageContract.getAssetListOfUser(addr);
+         // Total value of minted assets
+         uint256 totalValueMinted = 0;
+         //Sum up total value of minted assets
+         for(uint i = 0; i < assetsMinted.length; i++) {
+             uint256 assetAmount = storageContract.getAssetAmountOfUser(addr, assetsMinted[i]);
+             uint256 assetPrice = storageContract.getAssetPrice(assetsMinted[i]);
+             uint256 assetValue = assetAmount * assetPrice;
+             totalValueMinted += assetValue;
+         }
+         //return collateral / totalValueMinted 
+         return collateralValue  / totalValueMinted;
```

Comment:
Check that this operation behaves properly. Maybe try different scenarios with a simple function:
```
function div(uint256 a, b) public view returns (uint256) {
  return a / b;
}
```